synapse/handlers/saml_handler.py

```diff
@@ -324,3 +303,53 @@ def saml_response_to_user_attributes(
             "mxid_localpart": localpart,
             "displayname": displayname,
         }
+
+    @staticmethod
+    def parse_config(config: dict):
+        """Parse the dict provided by the homeserver's config
+        Args:
+            config: A dictionary containing configuration options for this provider
+        Returns:
+            _SamlConfig: A custom config object
+        """
+
+        class _SamlConfig(object):
+            pass
+
+        saml_config = _SamlConfig()
+
+        # Handle deprecated options
+
+        # If mxid_source_attribute or mxid_mapping is defined in the deprecated location, use
+        # that instead for backwards compatibility
+        saml_config.mxid_source_attribute = config[
+            "old_mxid_source_attribute"
+        ] or config.get("mxid_source_attribute", "uid")
+
+        # If mxid_mapping is defined, use that instead for backwards compatibility
+        mapping_type = config["old_mxid_mapping"] or config.get(
+            "mxid_mapping", "hexencode"
+        )
+        try:
+            saml_config.mxid_mapper = MXID_MAPPER_MAP[mapping_type]
+        except KeyError:
+            raise ConfigError(
+                "saml2_config.user_mapping_provider.config: %s is not a valid "
+                "mxid_mapping value" % (mapping_type,)
+            )
+        return saml_config
+
+    @staticmethod
+    def get_required_saml_attributes(config: dict):
+        """Returns the required attributes of a SAML
+
+        Args:
+            config: A dictionary containing configuration options for this provider
```

Comment:
looks like you're calling it with a `_SamlConfig`, not a dict (which is a good thing)